source/tokens.rs

```diff
@@ -702,15 +728,402 @@ named_attr!(
     )
 );
 
+/// A token is a structure pairing a span to any data.
+#[derive(Debug, PartialEq)]
+pub struct Token<'a, T> {
+    /// Value of the token.
+    pub value: T,
+
+    /// The attached span of the value.
+    pub span : Span<'a>
+}
+
+/// A span is a set of meta information about a token.
+///
+/// The `Span` structure can be used as an input of the nom parsers.
+#[derive(Debug, PartialEq, Copy, Clone)]
+pub struct Span<'a> {
+    /// The offset represents the position of the slice relatively to
+    /// the input of the parser. It starts at offset 0.
+    pub offset: usize,
+
+    /// The line number of the slice relatively to the input of the
+    /// parser. It starts at line 1.
+    pub line  : u32,
+
+    /// The column number of the slice relatively to the input of the
+    /// parser. It starts at column 1.
+    pub column: u16,
```

Comment:
Minified PHP code is really rare. I have checked important projects, like WordPress, and the biggest file is 19k lines, no 65k. 65k is… huge :-p.